File: web_login_flutter/lib/models/register_user_order.dart
import 'package:equatable/equatable.dart';

class RegisterUserOrder extends Equatable {
  final String _email;
  final String _name;
  final String _surname;
  final String _password;

  const RegisterUserOrder({required name, required password, required email, required surname}) :
        _name = name,
        _password = password,
        _email = email,
        _surname = surname;

  String get name => _name;
  String get password => _password;
  String get surname => _surname;
  String get email => _email;

  @override
  List<Object> get props => [_name, _password, _surname, _email];

}

File: web-login-core/src/main/java/com/example/weblogincore/domain/model/user/RegisterUserOrder.java
package com.example.weblogincore.domain.model.user;

public class RegisterUserOrder {
    private String email;
    private String name;
    private String surname;
    private String password;

    public RegisterUserOrder(String email, String name, String surname, String password) {
        this.email = email;
        this.name = name;
        this.surname = surname;
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public String getName() {
        return name;
    }

    public String getSurname() {
        return surname;
    }

    public String getPassword() {
        return password;
    }
}
